oxts-sdk-gal-cpp/src/gal-cpp/gad.cpp

```diff
@@ -567,73 +567,172 @@ namespace OxTS
 		return GetLocVar();
 	}
 	//==============================================================================
-	// GadSpeed
-	GadSpeed::GadSpeed(uint8_t stream_id) : Gad(stream_id, GEN_TYPE::GEN_SPEED) {}
+	// GadWheelSpeed
+	GadWheelSpeed::GadWheelSpeed(uint8_t stream_id) : Gad(stream_id, GEN_TYPE::GEN_WSPEED) {}
 
 	// val
-	void GadSpeed::SetSpeedFw(double speed)
+	void GadWheelSpeed::SetWheelSpeedFw(double wheelspeed)
 	{
 		SetDataMode(0);
 		SetDataValType(SPEED_SYS_TYPE::SPEED_SYS_FW_VEL);
-		SetDataVal(speed, 1.0, 0.0);
+		SetDataVal(wheelspeed, 1.0, 0.0);
 	}
 
-	double GadSpeed::GetSpeedFw() const
+	double GadWheelSpeed::GetWheelSpeedFw() const
 	{
 		return GetDataVal()[0];
 	}
 
-	void GadSpeed::SetSpeedFwVar(double v_s)
+	void GadWheelSpeed::SetWheelSpeedFwVar(double v_s)
 	{
 		SetDataVarSingle(v_s);
 	}
 
-	double GadSpeed::GetSpeedFwVar() const
+	double GadWheelSpeed::GetWheelSpeedFwVar() const
 	{
 		return GetDataVar()[0];
 	}
 
-	void GadSpeed::SetWheelspeedCount(double count, double period)
+	void GadWheelSpeed::SetWheelSpeedCount(double count, double period)
 	{
 		SetDataMode(0);
 		SetDataValType(SPEED_SYS_TYPE::SPEED_SYS_FW_VEL);
 		SetDataVal(count, period, 0.0);
 	}
 
-	std::vector<double> GadSpeed::GetWheelspeedCount() const
+	std::vector<double> GadWheelSpeed::GetWheelSpeedCount() const
 	{
 		return std::vector<double>{GetDataVal()[0], GetDataVal()[1]};
 	}
 
-	void GadSpeed::SetWheelspeedVar(double v_c)
+	void GadWheelSpeed::SetWheelSpeedVar(double v_c)
 	{
 		SetDataVarSingle(v_c);
 	}
 
-	double GadSpeed::GetWheelspeedVar() const
+	double GadWheelSpeed::GetWheelSpeedVar() const
 	{
 		return GetDataVar()[0];
 	}
 
 	// loc 
-	void GadSpeed::SetAidingLeverArmFixed(double x, double y, double z)
+	void GadWheelSpeed::SetAidingLeverArmFixed(double x, double y, double z)
 	{
 		SetLocMode(LOC_SYS::LOC_FIXED);
 		SetLocVal(x, y, z);
 	}
 
-	void GadSpeed::SetAidingLeverArmConfig()
+	void GadWheelSpeed::SetAidingLeverArmConfig()
 	{
 		SetLocMode(LOC_SYS::LOC_CONFIG);
 		SetLocVal(0.0, 0.0, 0.0);
 	}
 
-	void GadSpeed::SetAidingLeverArmOptimising()
+	void GadWheelSpeed::SetAidingLeverArmOptimising()
 	{
 		SetLocMode(LOC_SYS::LOC_KF);
 		SetLocVal(0.0, 0.0, 0.0);
 	}
 
+	std::vector<double> GadWheelSpeed::GetAidingLeverArm() const
+	{
+		return GetLocVal();
+	}
+	void GadWheelSpeed::SetAidingLeverArmVar(double x, double y, double z)
+	{
+		SetLocVarDiag(x, y, z);
+	}
+
+	std::vector<double> GadWheelSpeed::GetAidingLeverArmVar() const
+	{
+		return GetLocVar();
+	}
+	//==============================================================================
+	// GadSpeed
+	GadSpeed::GadSpeed(uint8_t stream_id) : Gad(stream_id, GEN_TYPE::GEN_SPEED) {}
+
+	// val
+	void GadSpeed::SetSpeedFwMs(double speed)
+	{
+		SetDataMode(0);
+		SetDataValType(SPEED_SYS_TYPE::SPEED_SYS_FW_VEL);
+		SetDataVal(speed, 1.0, 1.0);
```

Comment:
Default period of 1.0s seems high, though am happy to be corrected on this.

Reply:
As period is here used just for error calcs, this is what the base error model is built on in blended by default. Could be changed in future to be more representative.